src/lib/main.ts

```diff
@@ -39,5 +40,6 @@ instantsearch.createInfiniteHitsSessionStorageCache = createInfiniteHitsSessionS
 instantsearch.highlight = helpers.highlight;
 instantsearch.snippet = helpers.snippet;
 instantsearch.insights = helpers.insights;
+instantsearch.middleware = middleware;
```

Comment:
middleware vs middlewares

Reply:
"middlewares" is not proper English.

Reply:
https://en.wiktionary.org/wiki/middleware
![image](https://user-images.githubusercontent.com/499898/88528771-581b0f80-cfff-11ea-8aae-9d44f18e4e63.png)

I don't think I've seen `s` at the end of `middleware` before, though.
So you agree it's natural to have `middleware` although we have `routers, stateMappings, connectors, widgets`, right?

Reply:
I think it should be middlewares, at least for the object containing all possible middlewares. Interestingly my spell-checker only accepts singular "middleware" as correct, and not "middlewares"

Reply:
![image](https://user-images.githubusercontent.com/499898/91051486-7be26d00-e620-11ea-9c08-8df90f99ad3b.png)

https://www.wordhippo.com/what-is/the-plural-of/middleware.html

While `middleware` is more common, it's also safe to use `middlewares` if we want to explicitly indicate it's a collection.

Reply:
https://github.com/algolia/instantsearch.js/pull/4446/commits/f34170d98118f82c2975bb1f6aad23ec0cc1f3b9